httpgrpc/io.go

```diff
@@ -87,10 +87,18 @@ func readProtoMessage(in io.Reader, codec encoding.CodecV2, sz int32, m interfac
 }
 
 // asMetadata converts the given HTTP headers into GRPC metadata.
-func asMetadata(header http.Header) (metadata.MD, error) {
+// A HeaderMatcherFunc can be provided to filter or modify the headers.
+func asMetadata(header http.Header, matcher HeaderMatcherFunc) (metadata.MD, error) {
 	// metadata has same shape as http.Header,
 	md := metadata.MD{}
 	for k, vs := range header {
+		if matcher != nil {
+			var keep bool
+			if k, vs, keep = matcher(k, vs); !keep {
```

Comment:
I don't think you need the bool..
```suggestion
			if k, vs, keep = matcher(k, vs); len(vs) > 0 {
```

Reply:
Actually you can just
```suggestion
			k, vs = matcher(k, vs)
```
The below loop is a no-op if there are no values.